app/models/namespace_group_link.rb

```diff
@@ -20,6 +20,10 @@ class NamespaceGroupLink < ApplicationRecord
               in: [Group.sti_name, Namespaces::ProjectNamespace.sti_name]
             }
 
+  scope :for_namespace_and_ancestors, lambda { |namespace = nil|
```

Comment:
**nitpick (blocking):** can we add in a new scope to only get the `not_expired` links. See `app/models/personal_access_tokens.rb` line 22. Also please add a test for this, to make sure expired links do not grant access.

Reply:
Updated and added tests in 8050d2289794cbe59b706c96105d6579b8aef320